genienlp/data_utils/iterator.py

```diff
@@ -100,6 +103,12 @@ def __next__(self):
                 longest_example = self.data_source[batch_of_indices[0]] # we sorted in descending order of length, so the first one is the longest
             else:
                 longest_example = new_example # this is the first element in batch, and therefore the longest
+            examples_size = self.batch_size_fn(longest_example)
+            if examples_size > self.batch_size:
+                logger.warning('Skipping an example larger than batch size. Consider increasing the batch size to avoid this warning')
+                i += 1
```

Comment:
`self.last_batch_start_index` should be incremented as well, otherwise, during prediction (when `shuffle_and_repeat` is False) we might go over the same example more than once.
But even after this change, this implementation significantly increases the probability of selecting the first example that is just small enough to fit in a batch. This is fine for testing purposes. Your previous implementation which called `_get_next_batch_start_index()` was better in this regard (after adding the missing `i+=1`) .